web/src/main/webapp/dataViews/tableView.ts

```diff
@@ -163,10 +177,57 @@ export class TableView extends TSViewBase implements IScrollTarget {
         tblAndBar.appendChild(this.htmlTable);
         tblAndBar.appendChild(this.scrollBar.getHTMLRepresentation());
 
+        this.initFindBar();
+
         this.messageBox = document.createElement("div");
         this.topLevel.appendChild(this.messageBox);
     }
 
+    private addCheckbox(stringDesc: string ): HTMLInputElement{
+        let div: HTMLElement = document.createElement("div");
+        div.innerHTML = "&nbsp; &nbsp;";
+        this.findBar.appendChild(div)
+        let label = document.createElement("label");
+        label.textContent = stringDesc;
+        this.findBar.appendChild(label);
+        let checkBox: HTMLInputElement = document.createElement("input");
+        checkBox.type = "checkbox";
+        this.findBar.appendChild(checkBox);
+        let divEnd: HTMLElement = document.createElement("div");
+        divEnd.innerHTML = "&nbsp; | &nbsp;";
+        this.findBar.appendChild(divEnd)
+        return checkBox;
+    }
+
+    private initFindBar(): void {
+        this.findBar = document.createElement("div");
+        this.findBar.style.margin = "2px";
+        this.findBar.style.backgroundColor = "yellow";
+        this.findBar.style.flexDirection = "row";
+        this.findBar.style.display = "none";
+        this.findBar.style.flexWrap = "nowrap";
+        this.findBar.style.justifyContent = "flex-start";
+        this.findBar.style.alignItems = "stretch";
+
+        this.findInputBox = document.createElement("input");
+        this.findBar.appendChild(this.findInputBox);
+
+        let nextButton = this.findBar.appendChild(document.createElement("button"));
+        nextButton.innerText = "Next";
+        nextButton.onclick = () => this.find(true);
+
+        let prevButton = this.findBar.appendChild(document.createElement("button"));
+        prevButton.innerText = "Previous";
+        prevButton.onclick = () => this.find(false);
+
+        this.substringsFindCheckbox = this.addCheckbox("Match substrings: ");
+        this.regexFindCheckbox = this.addCheckbox("Treat as regular expression:")
+        this.caseFindCheckbox = this.addCheckbox("Case sensitive: ")
+        this.startFromTopCheckbox = this.addCheckbox("Start searching from the top: ")
```

Comment:
I think we should turn this into a button which is disabled when you are at the top.